### PR TITLE
Esbjava 3698

### PR DESCRIPTION
**Pull request by ravindraranwala**

### CODE REVIEW

File: modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        loadESBConfigurationFromClasspath("artifacts" + File.separator + "ESB" + File.separator +
+                                          "json" + File.separator + "StockQuoteAPI.xml");
+
+        final String log4jSourceFile =
```
Comment (isururanawaka) [inline modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java]: what is the real use case of making final here ?

File: modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                                       System.getProperty("framework.resource.location") +
+                                               "/artifacts/ESB/json/log4j.properties";
+        final String log4jDestinationFile =
+                                            CarbonBaseUtils.getCarbonHome() +
+                                                    "/repository/conf/log4j.properties";
+
+        serverConfigurationManager.applyConfiguration(new File(log4jSourceFile),
```
Comment (isururanawaka) [inline modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java]: What will happen if log4jSourceFile is null

File: modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        final HttpPost post = new HttpPost("http://localhost:8280/stockquote/test");
+        post.addHeader("Content-Type", "application/json");
+        post.addHeader("SOAPAction", "urn:getQuote");
+        StringEntity se = new StringEntity(getPayload());
+
+        post.setEntity(se);
+        HttpResponse response = httpClient.execute(post);
+
+        Thread.sleep(2000);
```
Comment (isururanawaka) [inline modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java]: Can we just throw interrupted Exception best way to handle is Thread.currentThread().interrupt();

File: modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java (hunk 1 above)
Comment (ravindraranwala) [inline modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java]: Making it final will prevent any inadvertent modifications to the field.

**Pull request closed without merging**